python/text-to-speech/main.py

```diff
@@ -0,0 +1,292 @@
+            "Ocp-Apim-Subscription-Key": subscription_key
+        }
+        # Send request with subscription key.
+        response = requests.post(fetch_token_url, headers=headers, timeout=10)
+        # Grab access token valid for 10 minutes.
+        access_token = response.text
+        return access_token
+
+    def speech(self, text: str, language: str) -> bytes:
+        """
+        Converts the given text into speech with the Google text to speech API.
+
+        Input:
+            text: The text to be converted into speech.
+            language: The language code (BCP-47 format).
+
+        Returns:
+            bytes: The synthezied speech in bytes.
+        """
+        # Endpoint for cognitive services speech api
+        url = (
+            f"https://{Azure.REGION}.tts."
+            "speech.microsoft.com/cognitiveservices/v1"
+        )
+        # Headers and auth for request.
+        headers_azure = {
+            "Content-type": "application/ssml+xml",
+            "Authorization": "Bearer " + self.get_token(self.api_key),
+            "X-Microsoft-OutputFormat": "audio-16khz-32kbitrate-mono-mp3",
+        }
+        data_azure = (
+            f"<speak version='1.0' xml:lang='{language}'><voice "
+            f"xml:lang='{language}' xml:gender='{Azure.GENDER}' "
+            f"name='{Azure.VOICE}'>{text}</voice></speak>"
+        )
+        response = requests.request(
+            "POST",
+            url,
+            headers=headers_azure,
+            data=data_azure,
+            timeout=10,
+        )
+        response.raise_for_status()
+        return response.content
+
+
+class AWS(TextToSpeech):
+    """Represent the implementation of AWS text to speech. """
+
+    VOICE_ID = "Joanna"
+
+    def validate_request(self, req: requests) -> None:
+        """
+        Validate the request data for AWS text to speech.
+
+        Input:
+            req (request): The request provided by the user.
+        Raises:
+            ValueError: If any required value is missing or invalid.
+        """
+        if not req.variables.get("API_KEY"):
+            raise ValueError("Missing API_KEY.")
+        if not req.variables.get("SECRET_API_KEY"):
+            raise ValueError("Missing SECRET_API_KEY.")
+        self.api_key = req.variables.get("API_KEY")
+        self.secret_api_key = req.variables.get("SECRET_API_KEY")
+
+    def speech(self, text: str, language: str) -> bytes:
+        """
+        Converts the given text into speech with the AWS text to speech API.
+
+        Input:
+            text: The text to be converted into speech.
+            language: The language code (BCP-47 format).
+
+        Returns:
+            bytes: The synthezied speech in bytes.
+        """
+        # Call polly client using boto3.session.
+        polly_client = boto3.Session(
+            aws_access_key_id=self.api_key,
+            aws_secret_access_key=self.secret_api_key,
+            region_name="us-west-2").client("polly")
```

Comment:
Do you want to make this REGION a class variable, similar to Azure?

Reply:
I turned it into a region class variables.